profiler/options.go

```diff
@@ -197,15 +198,28 @@ func WithAgentAddr(hostport string) Option {
 	}
 }
 
-// WithAPIKey is deprecated and might be removed in future versions of this
-// package. It allows to skip the agent and talk to the Datadog API directly
-// using the provided API key.
+// WithAPIKey is deprecated and will be ignored unless you also use the
```

Comment:
This is a breaking change IMHO. Removing this option vs. doing this - it's basically the same thing. I would ask you to consider logging a warning instead.

Reply:
@gbbr yes, it's a breaking change. There is more discussion about this in `PROF-2719`. 

The backend might eventually reject agentless-profile uploads, so this breaking change will happen regardless of making this change in the library or not.

IDK what semver has to say about "backend API changes", but clearly those kind of changes can often be outside of the control of the library.

I think this is the most user-friendly way to deal with this that won't force an immediate v2 release of dd-trace-go.

PS: [Everything](https://www.hyrumslaw.com/) is a breaking change 🙈 and semver doesn't have a lot to say about non-obvious breaking changes like this one. So for better or worse, every single change we make is a subtle tradeoff between "strict semver" less "toil" : /.

Reply:
Oh, this is a fun watch if you want to look further into the abyss of Hyrum's Law and how another part of Google has a totally different approach to tackling it compared to the Go/Semver folks: [CppCon 2017: Titus Winters “C++ as a "Live at Head" Language” - YouTube](https://www.youtube.com/watch?v=tISy7EJQPzI)

Reply:
What you're doing here:
* Making an existing (deprecated) option a no-op.
* Adding a new (deprecated out of the box) option to make the previous no-op option work.
What I'm proposing:
* Keep the original option
* Drop the new one
* Log a warning.